src/test/java/io/confluent/connect/jdbc/sink/JdbcDbWriterTest.java

```diff
@@ -71,6 +80,73 @@ private JdbcDbWriter newWriter(Map<String, String> props) {
     return new JdbcDbWriter(config, dialect, dbStructure);
   }
 
+  private JdbcDbWriter newWriterWithMockConnection(Map<String, String> props, Connection mockConnection) {
+    final JdbcSinkConfig config = new JdbcSinkConfig(props);
+    dialect = mock(DatabaseDialect.class);
+    final DbStructure dbStructure = mock(DbStructure.class);
+    return new JdbcDbWriter(config, dialect, dbStructure) {
+      protected CachedConnectionProvider connectionProvider(int maxConnAttempts, long retryBackoff) {
+        CachedConnectionProvider mockConnectionProvider = mock(CachedConnectionProvider.class);
+        when(mockConnectionProvider.getConnection()).thenReturn(mockConnection);
+        return mockConnectionProvider;
+      }
+    };
+  }
+
+  @Test
+  public void verifyConnectionRollback() throws SQLException {
```

Comment:
Please also keep the testcase where the rollback succeeded and did not throw. They can be consolidated with a helper method that takes a boolean argument indicating whether to set up the mock to throw an exception on rollback.